src/cleanupCommand/cleanupManager.ts

```diff
@@ -20,32 +22,48 @@ export class CleanupManager {
   ) {
     this.discreteBatchSize = config.get<number>('batch_size.discreteLayers');
     this.newIngestionJobType = config.get('new_ingestion_job_type');
+    this.updateIngestionJobType = config.get('update_ingestion_job_type');
+    this.sourceBlackList = config.get<string>('fs.blacklist_sources').split(',');
   }
 
   public async cleanFailedIngestionTasks(): Promise<void> {
     const FAILED_CLEANUP_DELAY = this.config.get<number>('failed_cleanup_delay_days.ingestion');
     const deleteDate = new Date();
     deleteDate.setDate(deleteDate.getDate() - FAILED_CLEANUP_DELAY);
-    const notCleanedAndFailed = await this.jobManager.getFailedAndNotCleanedIngestionJobs(this.newIngestionJobType);
 
-    for (let i = 0; i < notCleanedAndFailed.length; i += this.discreteBatchSize) {
-      const currentBatch = notCleanedAndFailed.slice(i, i + this.discreteBatchSize);
+    const notCleanedAndFailedNew = await this.jobManager.getFailedAndNotCleanedIngestionJobs(this.newIngestionJobType);
+    for (let i = 0; i < notCleanedAndFailedNew.length; i += this.discreteBatchSize) {
+      const currentBatch = notCleanedAndFailedNew.slice(i, i + this.discreteBatchSize);
       const expiredBatch = this.filterExpiredFailedTasks(currentBatch, deleteDate);
       if (expiredBatch.length > 0) {
-        await this.sourcesProvider.deleteDiscretes(expiredBatch);
+        const blackListFlitteredBatch = this.sourceBlackList.length > 0 ? this.filterBlackListSourcesTasks(expiredBatch) : expiredBatch;
+        await this.sourcesProvider.deleteDiscretes(blackListFlitteredBatch);
       }
       await this.tileProvider.deleteDiscretes(currentBatch);
       const failedDiscreteLayers = await this.mapproxy.deleteLayers(currentBatch);
       const completedDiscretes = expiredBatch.filter((el) => !failedDiscreteLayers.includes(el));
       await this.jobManager.markAsCompletedAndRemoveFiles(completedDiscretes);
     }
+
+    const notCleanedAndFailedUpdate = await this.jobManager.getFailedAndNotCleanedIngestionJobs(this.updateIngestionJobType);
```

Comment:
This whole logic repeats itself with `filterExpiredFailedTasks` too, maybe move this logic to a function or at least a dedicated loop?

Reply:
the follwoing section only "deleteExpiredBatch(.."
![image](https://user-images.githubusercontent.com/18442671/214076998-b2ec9719-eccf-478c-a2e0-f04401849739.png)

Reply:
done